### PR TITLE
Adds TransactionByHashFull method that returns "From"

### PR DESCRIPTION
**Pull request by treeder**

### CODE REVIEW

File: goclient/goclient.go [hunk 1]
```diff
@@ -199,12 +199,36 @@ func (ec *Client) TransactionByHash(ctx context.Context, hash common.Hash) (tx *
 	} else if _, r, _ := json.tx.RawSignatureValues(); r == nil {
 		return nil, false, fmt.Errorf("server returned transaction without signature")
 	}
+	fmt.Printf("json: %+v\n", json.From)
```
Comment (treeder) [inline goclient/goclient.go:202]:
```suggestion
```

**Comment by treeder** [comment]
That requires two separate RPC calls though, seems kind of wasteful. What if make this new function something like TransactionByHashWithSender and return a new object that extends/embed types.Transaction, but with the extra field. 

eg:

```go
type TransactionWithSender struct {
	types.Transaction
	From *common.Address `json:"from,omitempty"`
}
```

**Comment by jmank88** [comment]
> That requires two separate RPC calls though

It doesn't:

```
// There is a fast-path for transactions retrieved by TransactionByHash and
// TransactionInBlock. Getting their sender address can be done without an RPC interaction.
```

**Comment by treeder** [comment]
I see it says that, but is this not an RPC call? https://github.com/gochain/gochain/blob/master/goclient/goclient.go#L224

**Comment by jmank88** [comment]
That doesn't execute if it already has it https://github.com/gochain/gochain/blob/eb76d2c6205f7e53f139b3f66c68c7554e2ef2d5/goclient/goclient.go#L215

**Comment by treeder** [comment]
Do you know how to get the block hash from a transaction?

**Comment by treeder** [comment]
Updated it to feel like the original method TransactionByHash method, but with extra fields.

**Review by jmank88 — APPROVED** [review]
LGTM as long as embedding the unexported type isn't a problem.

The `isPending bool` return var is redundant now since the caller can check themselves, so could be dropped.

**Comment by treeder** [comment]
Seems to work fine in use (already using it). 

I'll leave `isPending` just to keep the same syntax if someone wants to switch to this.

**Pull request merged**